modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/hooks/useResource.es.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {useResource} from '@clayui/data-provider';
+
+import {useStorage} from './useStorage.es';
+
+const ENDPOINT_FIELD_TYPES = `${window.location.origin}/o/dynamic-data-mapping-form-field-types`;
+
+const HEADERS = {
+	Accept: 'application/json',
+	'Accept-Language': Liferay.ThemeDisplay.getBCP47LanguageId(),
+	'Content-Type': 'application/json',
+};
+
+export const useFieldTypesResource = () => {
```

Comment:
Looking forward to being able to rely on `fetch` from `frontend-js-web` here. 😁

Reply:
haha yeah, I'm waiting for https://github.com/liferay/clay/pull/3417 🤪. This is very dependent on the cache.